File: backend/src/main/java/com/echonovel/service/AuthService.java
package com.echonovel.service;

import com.echonovel.dto.request.LoginRequest;
import com.echonovel.dto.request.RegisterRequest;
import com.echonovel.dto.response.AuthResponse;

/**
 * Service interface for authentication operations.
 */
public interface AuthService {

    /**
     * Send OTP for registration
     */
    void sendRegisterOtp(com.echonovel.dto.request.SendOtpRequest request);

    /**
     * Register a new member account.
     */
    AuthResponse register(RegisterRequest request);

    /**
     * Login with email and password.
     */
    AuthResponse login(LoginRequest request);

    /**
     * Send OTP for forgot password
     */
    void sendForgotPasswordOtp(com.echonovel.dto.request.ForgotPasswordRequest request);

    /**
     * Reset password with OTP
     */
    void resetPassword(com.echonovel.dto.request.ResetPasswordRequest request);

    /**
     * Google Login/Register
     */
    AuthResponse googleLogin(com.echonovel.dto.request.GoogleAuthRequest request);

    /**
     * Refresh Access Token
     */
    AuthResponse refreshAccessToken(com.echonovel.dto.request.RefreshTokenRequest request);

    /**
     * Logout
     */
    void logout(com.echonovel.dto.request.LogoutRequest request);
}

File: frontend/src/services/authService.ts
import api from './api';
import type { LoginRequest, RegisterRequest, AuthResponse, ApiResponse } from '../types';

export const authService = {
  login: async (data: LoginRequest): Promise<ApiResponse<AuthResponse>> => {
    const response = await api.post<ApiResponse<AuthResponse>>('/auth/login', data);
    return response.data;
  },

  register: async (data: RegisterRequest): Promise<ApiResponse<AuthResponse>> => {
    const response = await api.post<ApiResponse<AuthResponse>>('/auth/register', data);
    return response.data;
  },

  sendRegisterOtp: async (data: { email: string }): Promise<ApiResponse<void>> => {
    const response = await api.post<ApiResponse<void>>('/auth/register/send-otp', data);
    return response.data;
  },

  sendForgotPasswordOtp: async (data: import('../types').ForgotPasswordRequest): Promise<ApiResponse<void>> => {
    const response = await api.post<ApiResponse<void>>('/auth/forgot-password/send-otp', data);
    return response.data;
  },

  resetPassword: async (data: import('../types').ResetPasswordRequest): Promise<ApiResponse<void>> => {
    const response = await api.post<ApiResponse<void>>('/auth/forgot-password/reset', data);
    return response.data;
  },

  googleLogin: async (data: import('../types').GoogleAuthRequest): Promise<ApiResponse<AuthResponse>> => {
    const response = await api.post<ApiResponse<AuthResponse>>('/auth/google', data);
    return response.data;
  },

  refreshToken: async (data: import('../types').RefreshTokenRequest): Promise<ApiResponse<AuthResponse>> => {
    const response = await api.post<ApiResponse<AuthResponse>>('/auth/refresh', data);
    return response.data;
  },

  logout: async (data: import('../types').LogoutRequest): Promise<ApiResponse<void>> => {
    const response = await api.post<ApiResponse<void>>('/auth/logout', data);
    return response.data;
  },
};
